### PR TITLE
[vcpkg ci] error out cmake related deprecated functions

### PR DESCRIPTION
**Pull request by JackBoosY**
Throw error message when cmake-related deprecated function is detected in the changes.

Related: https://github.com/microsoft/vcpkg/pull/30070

### CODE REVIEW

**Comment by JackBoosY** [comment]
Okay I need to pick another var name.

**Comment by Adela0814** [comment]
Thanks for your PR.
Sometimes additional problems arise after deprecated function changes, which may affect the merge of high-priority fixes. Our guidebook says to avoid using deprecated functions, not prohibit.  @BillyONeal  Please help to confirm whether we need to error out the deprecated function?

**Comment by BillyONeal** [comment]
> Sometimes additional problems arise after deprecated function changes, which may affect the merge of high-priority fixes. Our guidebook says to avoid using deprecated functions, not prohibit. @BillyONeal Please help to confirm whether we need to error out the deprecated function?

The maintainer guidelines describe it as a suggestion because we did not want to force people to make unrelated changes for simple version updates and similar. There was still a lot of use of the deprecated function in the curated registry, and forcing people who aren't interested in overall 'cleanup' of code they don't own in our registry makes it less nice to be a contributor.

Now that there are no calls to the deprecated functions left in the tree, I think we can block it entirely.

Note that CI doesn't provide a means of testing this change because it only runs some of these scripts from master rather than from the PR branch.

**Review by Adela0814 — DISMISSED** [review]

**Comment by dg0yt** [comment]
> Note that CI doesn't provide a means of testing this change because it only runs some of these scripts from master rather than from the PR branch.

@BillyONeal Can't you trigger a manual run in the vcpkg.ci pipeline (no parent hashes, no caching)?

**Comment by BillyONeal** [comment]
> @BillyONeal Can't you trigger a manual run in the vcpkg.ci pipeline (no parent hashes, no caching)?

I can trigger manual runs of the Azure Pipelines stuff against forks, but not Actions stuff against forks. (Because Actions can write-back to the repo and Pipelines can't, Actions only ever runs `trustedPR.yml` from *master*, not from the fork)

To meaningfully test changes to the Actions stuff one needs to setup Actions in their own fork with `main` pointing to the base commit SHA of the PR, and then make PRs into it intentionally triggering problems to show that they are problems.

**Review by BillyONeal — DISMISSED** [review]
This causes the bot to break. See test cases:

* Bad deprecated function: https://github.com/BillyONeal/vcpkg/pull/2 (Bot crashes)
* Bad cmake: https://github.com/BillyONeal/vcpkg/pull/4/ (This one works)
* Good: https://github.com/BillyONeal/vcpkg/pull/3 (Bot crashes)

**Comment by Adela0814** [comment]
@JackBoosY  Please reactivate this PR once it is ready for review. Thanks.

**Comment by JackBoosY** [comment]
@BillyONeal Please retest CI scripts.

File: .github/workflows/untrustedPR.yml [hunk 1]
```diff
@@ -92,34 +94,55 @@ jobs:
...
+              let deprecated_functions = {
+                  vcpkg_extract_source_archive_ex: 'vcpkg_extract_source_archive https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_extract_source_archive',
+                  vcpkg_build_msbuild: 'vcpkg_install_msbuild https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_install_msbuild',
+                  vcpkg_apply_patches: 'the PATCHES arguments to the \"extract\" helpers (for example, vcpkg_from_github() (https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_from_github))',
+              };
+              for (let line of deprecated_function) {
+                // line has the format: <path>:<line_number>:<matched_string
+                // For example: ports/rtlsdr/portfile.cmake:42:    vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/${PORT})
+                [file, line_number, match] = line.split(':');
+                for (deprecated_function in deprecated_functions) {
```
Comment (BillyONeal) [inline .github/workflows/untrustedPR.yml:112]: This still crashed when @dan-shaw and I tested again.

I think the problem is that this is trying to write to `deprecated_function` which is defined to be `const` on line 57.

Comment (JackBoosY) [reply .github/workflows/untrustedPR.yml:112]: make an appropriate name is always a hard work.

**Comment by dan-shaw** [comment]
Can you test this on your repository with an Actions pipeline? See Billy's comment on how to do so

**Comment by JackBoosY** [comment]
Looks good:
- https://github.com/microsoft/vcpkg/pull/31692
- https://github.com/microsoft/vcpkg/pull/31690
- https://github.com/microsoft/vcpkg/pull/31691

**Review by dan-shaw — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**